camel/toolkits/jina_reranker_toolkit.py

```
@@ -30,47 +37,70 @@ class JinaRerankerToolkit(BaseToolkit):
     def __init__(
         self,
         timeout: Optional[float] = None,
+        model_name: Optional[str] = "jina-reranker-m0",
         device: Optional[str] = None,
+        use_api: bool = True,
     ) -> None:
         r"""Initializes a new instance of the JinaRerankerToolkit class.
 
         Args:
             timeout (Optional[float]): The timeout value for API requests
                 in seconds. If None, no timeout is applied.
                 (default: :obj:`None`)
+            model_name (Optional[str]): The reranker model name. If None,
+                will use the default model.
+                (default: :obj:`None`)
             device (Optional[str]): Device to load the model on. If None,
                 will use CUDA if available, otherwise CPU.
                 (default: :obj:`None`)
+            use_api (bool): A flag to switch between local model and API.
+                (default: :obj:`True`)
         """
-        import torch
-        from transformers import AutoModel
 
         super().__init__(timeout=timeout)
 
-        self.model = AutoModel.from_pretrained(
-            'jinaai/jina-reranker-m0',
-            torch_dtype="auto",
-            trust_remote_code=True,
-        )
-        DEVICE = (
-            device
-            if device is not None
-            else ("cuda" if torch.cuda.is_available() else "cpu")
-        )
-        self.model.to(DEVICE)
-        self.model.eval()
+        self.use_api = use_api
+        self.model_name = model_name
+
+        if self.use_api:
+            self.model = None
+            self._api_key = os.environ.get("JINA_API_KEY", "None")
+            if self._api_key == "None":
+                logger.error("JINA_API_KEY environment variable is not set.")
```

Maybe better than a error logger, how about raising a valueerror (or similar) to make the process fail faster?